### PR TITLE
Rename heat conduction to heat transfer

### PR DESCRIPTION
**Pull request by GiudGiud**
let s see how the apps do
the one tedious thing could be if they link to doco pages in the module

refs https://github.com/idaholab/moose/issues/25584

Recipe patches needed:
https://github.com/idaholab/civet_recipes/pull/1462
- Framework 1
- Modules
- Modules parallel
- Modules recover
- Install combined

App patches needed:
- [x] Marmot https://github.inl.gov/ncrc/marmot/pull/910
- [x] Bison https://github.inl.gov/ncrc/bison/pull/5619
- [x] Grizzly https://github.inl.gov/ncrc/grizzly/pull/742
- [x] Pronghorn https://github.inl.gov/ncrc/pronghorn/pull/1224
- [x] Blue Crab https://github.inl.gov/ncrc/blue_crab/pull/224
- [x] Direwolf https://github.inl.gov/ncrc/dire_wolf/pull/282
- [x] Griffin https://github.inl.gov/ncrc/griffin/pull/1526
- [x] Relap7 https://github.inl.gov/ncrc/relap-7/pull/2548
- [x] SAM https://git-nse.egs.anl.gov/SAM/SAM/-/merge_requests/828
- [x] Sockeye https://github.inl.gov/ncrc/sockeye/pull/728
- [x] BlackBear https://github.com/idaholab/blackbear/pull/361
- [x] Isopod https://github.com/idaholab/isopod/pull/83
- [x] Magpie https://github.com/idaholab/magpie/pull/490
- [x] Malamute https://github.com/idaholab/malamute/pull/104
- [x] Mastodon https://github.com/idaholab/mastodon/pull/437
- [x] TMAP8 https://github.com/idaholab/TMAP8/pull/72

### CODE REVIEW

**Comment by GiudGiud** [comment]
I think we should get this in like:
- get the green light from a reviewer here
- merge the civet recipe change (which breaks next already)
- make sure it s green on those
- merge this PR (unbreaks previous break, breaks due to apps)
- merge all the app patches

File: modules/heat_transfer/doc/content/modules/heat_transfer/sqa/heat_transfer_far.md [hunk 1]
```diff
@@ -0,0 +1 @@
+!template load file=sqa/module_far.md.template category=heat_transfer module=Heat Conduction
```
Comment (lindsayad) [inline modules/heat_transfer/doc/content/modules/heat_transfer/sqa/heat_transfer_far.md:1]: Maybe you are missing "Heat Conduction" -> "Heat Transfer"?

Comment (GiudGiud) [reply modules/heat_transfer/doc/content/modules/heat_transfer/sqa/heat_transfer_far.md:1]: good catch

Comment (GiudGiud) [reply modules/heat_transfer/doc/content/modules/heat_transfer/sqa/heat_transfer_far.md]: fixed

**Comment by loganharbour** [comment]
This has a bit to go. Still need to do the makefile changes, and should probably be able to get away with not immediately breaking everyone by setting up a symlink in modules

**Comment by GiudGiud** [comment]
All the apps build properly. We only break their documentation because the sqa docs are all hardcoded. 

I changed a lot of makefiles already. I don’t know which one is missing changes

**Comment by GiudGiud** [comment]
Making symlinks to the heat_conduction module got 3 more recipes green.
I think we can proceed with the 5-step plan now

File: modules/heat_conduction/include
Comment (lindsayad) [inline modules/heat_conduction/include:1]: Shouldn't we just need the top-level symlink?

File: modules/heat_transfer/src/base/HeatConductionApp.C [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "HeatTransferApp.h"
```
Comment (lindsayad) [inline modules/heat_transfer/src/base/HeatConductionApp.C:10]: Seems like this file is unnecessary. But if it's necessary
```suggestion
#include "HeatConductionApp.h"
```

File: modules/heat_transfer/test/include/base/HeatConductionTestApp.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseApp.h"
+#include "HeatTransferTestApp.h"
+
+class HeatTransferTestApp : public HeatConductionTestApp
```
Comment (lindsayad) [inline modules/heat_transfer/test/include/base/HeatConductionTestApp.h:15]:
```suggestion
class HeatConductionTestApp : public HeatTransferTestApp
```

File: modules/module_loader/include/ModulesApp.h [hunk 4]
```diff
@@ -35,6 +35,9 @@
 #ifdef HEAT_CONDUCTION_ENABLED
 #include "HeatConductionApp.h"
 #endif
+#ifdef HEAT_TRANSFER_ENABLED
+#include "HeatTransferApp.h"
+#endif
```
Comment (lindsayad) [inline modules/module_loader/include/ModulesApp.h:40]: So is this part of the 5 step plan having these both together?

File: modules/module_loader/include/ModulesApp.h [hunk 5]
```diff
@@ -135,8 +138,8 @@ ModulesApp::registerAllObjects(Factory & f, ActionFactory & af, Syntax & s)
   GeochemistryApp::registerAll(f, af, s);
 #endif
 
-#ifdef HEAT_CONDUCTION_ENABLED
-  HeatConductionApp::registerAll(f, af, s);
```
Comment (lindsayad) [inline modules/module_loader/include/ModulesApp.h:139]: Seems like if you are going to keep the above lines, then you should keep these as well

File: modules/module_loader/src/ModulesApp.C
Comment (lindsayad) [inline modules/module_loader/src/ModulesApp.C:1]: My same questions from the header apply here

File: modules/modules.mk [hunk 6]
```diff
@@ -62,19 +62,25 @@ endif
...
-        HEAT_CONDUCTION             := yes
+        HEAT_TRANSFER               := yes
 endif
 
 ifeq ($(CONTACT),yes)
         TENSOR_MECHANICS            := yes
 endif
 
+# heat_conduction was renamed to heat_transfer
 ifeq ($(HEAT_CONDUCTION),yes)
```
Comment (lindsayad) [inline modules/modules.mk:78]: Is it possible for this to be yes considering your line removals above?

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: modules/heat_conduction/include
Comment (GiudGiud) [reply modules/heat_conduction/include:1]: If you do that you run the tests twice in the same folders and that crashes the 3 recipes I mentione

File: modules/modules.mk (hunk 6 above)
Comment (GiudGiud) [reply modules/modules.mk:78]: if set externally

File: modules/module_loader/include/ModulesApp.h (hunk 5 above)
Comment (GiudGiud) [reply modules/module_loader/include/ModulesApp.h:139]: this code is performed by inheritance we dont need to redefine it

File: modules/heat_transfer/src/base/HeatConductionApp.C (hunk 2 above)
Comment (GiudGiud) [reply modules/heat_transfer/src/base/HeatConductionApp.C]: yeah that s what I was trying to do
I want that header to be included at least once so it s compiled

File: modules/module_loader/include/ModulesApp.h (hunk 4 above)
Comment (GiudGiud) [reply modules/module_loader/include/ModulesApp.h:40]: the 3 lines above those will be removed

File: modules/modules.mk (hunk 6 above)
Comment (GiudGiud) [reply modules/modules.mk:78]: also app makefiles still refer to heat_conduction in their makefile

**Review by lindsayad — DISMISSED** [review]
This looks ok to me, but it would be nice to have integration sign-off from @loganharbour

**Review by loganharbour — DISMISSED** [review]
- Need to fix the installed dirs for combined, see https://civet.inl.gov/job/1795152/
- recipe PR isn't passing

**Comment by GiudGiud** [comment]
The installed dirs for combined was fooled by the symlink

looks like civet_recipe is hosed right now. My PR was passing fine

**Comment by loganharbour** [comment]
yeah, the SSH issues to github are also on HPC. definitely best to wait for that to be resolved

**Comment by loganharbour** [comment]
There's an env update that's been stuck on next, which definitely needs to get to moose master and then to all app masters. Next hasn't merged in a bit. I would definitely suggest waiting for all of that to propagate through before this train. At least fully through the evening, assuming the SSH issues get resolved. good otherwise

**Comment by GiudGiud** [comment]
Sounds good. 

Rebased to handle conflict

CIVET recipes work again (green on testing)
Next has merged two hours ago https://civet.inl.gov/job/1796527/

**Comment by loganharbour** [comment]
SSH access to github.com is still broken, merge events on civet are still very flaky. talk to @milljm about the ticket being completed

**Review by lindsayad — CHANGES_REQUESTED** [review]
blocking while we get feedback from the community

**Review by lindsayad — DISMISSED** [review]

**Comment by GiudGiud** [comment]
Just got an idea that might reduce app failures. Let s see.
It s essentially a double symlink

**Comment by GiudGiud** [comment]
no luck on fooling the apps doco builds

**Comment by GiudGiud** [comment]
Let's go?

**Comment by lindsayad** [comment]
Based on slack it sounds like @loganharbour wants to look at the CI results

**Review by lindsayad — DISMISSED** [review]

**Comment by GiudGiud** [comment]
I'll wait for github.inl.gov & co (seems there s a wednesday thing too) to be back in working order but thanks for the approval!

**Comment by loganharbour** [comment]
Why are the names in the makefiles not changed in the app patches?

**Comment by GiudGiud** [comment]
It still builds with the old name, we can have it either way during the transition phase.

**Comment by loganharbour** [comment]
> It still builds with the old name, we can have it either way during the transition phase.

why would we not do it all at once?

**Comment by GiudGiud** [comment]
I can do it when I update the submodule

**Review by loganharbour — APPROVED** [review]

**Comment by loganharbour** [comment]
See message in slack. Good to go at 9 MST

**Pull request merged**